hare/eligibility/beacon.go

```diff
@@ -52,13 +55,21 @@ func (b *beacon) Value(layer types.LayerID) (uint32, error) {
 		return val.(uint32), nil
 	}
 
+	// note: multiple concurrent calls to ContextuallyValidBlock and calcValue can be made
+	// consider adding a lock if concurrency-optimized is important
 	v, err := b.patternProvider.ContextuallyValidBlock(sl)
 	if err != nil {
-		log.With().Error("Could not get pattern Id",
+		b.Log.With().Error("Could not get pattern Id",
 			log.Err(err), log.LayerId(uint64(layer)), log.Uint64("sl_id", uint64(sl)))
 		return nilVal, errors.New("could not calc beacon value")
 	}
 
+	// notify if there are no contextually valid blocks
+	if len(v) == 0 {
+		b.Log.With().Warning("hare beacon: zero contextually valid blocks (ignore if genesis first layers)",
```

Comment:
so maybe log this only inf not genesis ? if it's too hard to know if it's genesis then not important

Reply:
If everything is fine, then in layer 2 (tortoise eats layer 1 blocks so layer 0 becomes valid) we should stop getting this message, because blocks of `sl` (should be layer 0) are valid. So only in layer 1 we expect to ignore this message.

Reply:
Already discussed it with Barak, said it is enough (saves code complexity).

Reply:
Looking at the function m.ContextuallyValidBlock, it looks like this case is already handled and that you can remove the parenthesis